tests/runtime/http_server/test_http_server.py

```diff
@@ -1063,57 +1139,222 @@ def test_train_long_running_sample_task(client, runtime_http_server):
-    ):
-        with pytest.raises(ValueError):
-            http_server.RuntimeHTTPServer()
+    }
+    training_response = client.post(
+        "/api/v1/SampleTaskSampleModuleTrain",
+        json=json_input,
+    )
+
+    # Start the training and make sure it starts successfully
+    training_json_response = training_response.json()
+    training_response.raise_for_status()
+
+    # Get the status for the training
+    training_id = training_json_response["training_id"]
+    get_response = client.get(
+        http_server.TRAINING_MANAGEMENT_ENDPOINT, params={"training_id": training_id}
+    )
+    get_response.raise_for_status()
+
+
+def test_training_management_cancel(client):
+    """Test that trainings can be canceled"""
+    model_name = "sample_task_train"
+    json_input = {
+        "model_name": model_name,
+        "parameters": {
+            "training_data": {"data_stream": {"data": [{"number": 1}]}},
+            "batch_size": 42,
+            # Sleep the job so that cancellation can interrupt it. It will not
+            # sleep for this long in practice.
+            "sleep_time": 20,
+        },
+    }
+    training_response = client.post(
+        "/api/v1/SampleTaskSampleModuleTrain",
+        json=json_input,
+    )
+
+    # Start the training and make sure it starts successfully
+    training_json_response = training_response.json()
+    training_response.raise_for_status()
+
+    # Cancel the training
+    training_id = training_json_response["training_id"]
+    cancel_response = client.delete(
+        http_server.TRAINING_MANAGEMENT_ENDPOINT, params={"training_id": training_id}
+    )
+    cancel_response.raise_for_status()
+
+    # Make sure the status reflects being canceled
+    get_response = client.get(
+        http_server.TRAINING_MANAGEMENT_ENDPOINT, params={"training_id": training_id}
+    )
+    get_response.raise_for_status()
+    assert get_response.json()["state"] == "CANCELED"
+
+
+def test_training_management_errors(client):
+    """Test that training status can be retrieved"""
+    model_name = "sample_task_train"
+    json_input = {
+        "model_name": model_name,
+        "parameters": {
+            "training_data": {"data_stream": {"data": [{"number": 1}]}},
+            "batch_size": 42,
+        },
+    }
+    training_response = client.post(
+        "/api/v1/SampleTaskSampleModuleTrain",
+        json=json_input,
+    )
+
+    # Make sure unknown training GET returns 404
+    assert (
+        client.get(
+            http_server.TRAINING_MANAGEMENT_ENDPOINT, params={"training_id": "bad-id"}
+        ).status_code
+        == 404
+    )
+
+    # Make sure missing param returns 422
+    assert client.get(http_server.TRAINING_MANAGEMENT_ENDPOINT).status_code == 422
+
+    # Make sure unknown training DELETE returns 404
+    assert (
+        client.delete(
+            http_server.TRAINING_MANAGEMENT_ENDPOINT, params={"training_id": "bad-id"}
+        ).status_code
+        == 404
+    )
```

Comment:
![image](https://github.com/caikit/caikit/assets/1646872/359ab5b9-0128-4e29-bf02-6088b0a95a3a)

Not a total blocker and very related to my other comment about splitting up http servicers, but a few minutes splitting this file up on some logical boundaries would pay off a lot in readability

Reply:
On the one hand, I _totally_ agree that there should be a full overhaul of the modularity in `http_server`, and I've taken a first step in this PR, so I'm already ankles-deep. I'd prefer not to further bloat this feature PR with the full refactor though. Good to skip and revisit?